client/my-sites/themes/themes-selection.jsx

```diff
@@ -107,6 +108,26 @@ const ThemesSelection = React.createClass( {
 		this.props.incrementPage();
 	},
 
+	//intercept preview and add primary and secondary
+	getOptions( theme ) {
+		const options = this.props.getOptions( theme );
+		const wrappedPreviewAction = ( action ) => {
+			return ( themeObj ) => {
+				const defaultAction = ( theme.price && options.purchase ) ? options.purchase : this.props.defaultOption;
```

Comment:
Note (probably for a subsequent PR): We shouldn't tack on any additional logic to determine the default option here.

Apparently, we also shouldn't be setting the `defaultOption` prop on `connectOptions`-connected components ([example](https://github.com/Automattic/wp-calypso/blob/5c3443763f99c7158e0be0001bba1f6474600abf/client/my-sites/themes/multi-site.jsx#L39)). Right now, those are just string literals, but as this line ^ shows, we really need something more flexible.

I think we really should compute `primary` and `secondary` inside `connectOptions`'s `mapStateToProps` function, essentially by using [this logic](https://github.com/Automattic/wp-calypso/blob/5c3443763f99c7158e0be0001bba1f6474600abf/client/my-sites/theme/main.jsx#L591-L601) from the theme sheet.